state_machine/drivers/emulation/lib/reader.py

```diff
@@ -0,0 +1,15 @@
+import sys
+import select
+import json
+
+def read(cubesat):
+    # While there is something on stdin read it (non blocking)
+    while select.select([sys.stdin, ], [], [], 0.0)[0]:
+        data = sys.stdin.readline()
+        if len(data) > 3 and data[0:3] == ">>>":
+            # print(data)
+            cubesat.sim = True
+            if data[3] == 'ω':
+                cubesat._gyro = json.loads(data[4:])
```

Comment:
Is the data actually in JSON? If that is the case, shouldn't you be able to get the type (ω or b) from the JSON?

Reply:
Seems reasonable. Is this something that will transfer to hardware, or will we not have a JSON library?

Reply:
This is in the emulation driver, meaning that this code is meant purely to be running on someones computer, not the micro controller. If we plan to transfer it to the microcontroller (are we planning on this?) we will either use the json library if it's available and otherwise we will roll our own solution.